pkg/config/teeconfig/teeconfig.go

```diff
@@ -0,0 +1,409 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package teeconfig is a tee of two configs that writes to both but reads from only one
+package teeconfig
+
+import (
+	"io"
+	"strings"
+	"time"
+
+	"github.com/DataDog/viper"
+	"github.com/spf13/afero"
+
+	"github.com/DataDog/datadog-agent/pkg/config/model"
+)
+
+// teeConfig is a combination of two configs, both get written to but only baseline is read
+type teeConfig struct {
+	baseline model.Config
+	compare  model.Config
+}
+
+// NewTeeConfig constructs a new teeConfig
+func NewTeeConfig(baseline, compare model.Config) model.Config {
+	return &teeConfig{baseline: baseline, compare: compare}
+}
+
+// OnUpdate adds a callback to the list receivers to be called each time a value is changed in the configuration
+// by a call to the 'Set' method.
+// Callbacks are only called if the value is effectively changed.
+func (t *teeConfig) OnUpdate(callback model.NotificationReceiver) {
+	t.baseline.OnUpdate(callback)
```

Comment:
Should we add the `OnUpdate` to the `compare` as well?

Reply:
Good catch, fixed